### PR TITLE
Feature simple http server

### PR DESCRIPTION
**Pull request by vivienschilis**
Hey man,

What do you thing of this,
A simple way to create a socketmaster compatible server

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	a.listener = NewTrackingListener(l)
+	return nil
+}
+
+func (a *app) signalHandler() {
+
+	c := make(chan os.Signal)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT, syscall.SIGHUP)
+	go func(c chan os.Signal, listener net.Listener) {
+		for {
```
Comment (zimbatm) [inline server/server.go]: I think the function could just return after the first signal

File: server/server.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+			<-c // os.Signal
+			log.Println("Closing listener")
```
Comment (zimbatm) [inline server/server.go]: Maybe let the user of the library pass a logger so he can hide those messages if he wants.

**Comment by zimbatm** [comment]
A nice implementation of supporting code. I'm happy to merge if you put it under clients/go or something ? Right now it's a bit confusing whenever it's part of socketmaster or not.

**Comment by vivienschilis** [comment]
@zimbatm you mean rename the package server into a package client ?

File: server/server.go (hunk 1 above)
Comment (vivienschilis) [inline server/server.go]: :+1:

**Comment by zimbatm** [comment]
You're right, "client" is not the right term. What about "github.com/zimbatm/socketmaster/slave" ?

```
slave.Listen()
slave.NewHttpServer()
slave.Server()
```

**Comment by vivienschilis** [comment]
sounds good

**Comment by vivienschilis** [comment]
@zimbatm updated

**Pull request merged**